programs/account-compression/src/instructions/append_leaves.rs

```diff
@@ -69,22 +69,24 @@ pub fn process_append_leaves_to_merkle_trees<'a, 'b, 'c: 'info, 'info>(
     let mut changelog_events = Vec::new();
```

Comment:
We still have a vector here that will get big.
We should try whether the optimizations in this pr are enough or we need to do more.


To further optimize we could:
1. get the changelog index of every Merkle tree in the beginning of the instruction
2. execute the appends in a heap neutral function and return nothing
3. use get_changelog_event to get the events
4. emit the event

Reply:
It probably makes sense to profile heap memory usage first. In case that appends only operate on the stack the above is not going to yield any benefits.

Reply:
I don't think you can avoid allocating **any** vector without changing the `struct Changelogs` definition. If we still want to keep the same format of logs, we need to allocate such vector at some point. I think the realistic goal to reach is making sure we do it only once in the whole program.

Otherwise, if we want to avoid allocating vectors in general, we could emit each single changelog in log_wrapper. Would Helius be fine with that? If yes, I'm happy to do that. If no, then again - I think the best we can do is ensuring we instantiate the vector only once.